core/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -254,8 +338,10 @@ public final XContentParser contentParser() throws IOException {
         BytesReference content = content();
         if (content.length() == 0) {
             throw new ElasticsearchParseException("Body required");
+        } else if (xContentType.get() == null) {
+            throw new IllegalStateException("Content-Type must be provided");
```

Comment:
this is a bug right? cause we already auto-detected if we didn't get the header? But then the error message is misleading I think